.github/workflows/python.yml

```diff
@@ -11,20 +11,17 @@ on:
 
 jobs:
   test:
-    container:
-      image: python:${{ matrix.config.version }}
     runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.config.experimental }}
     strategy:
       matrix:
-        config:
-        - { version: 3.6, experimental: false }
-        - { version: 3.7, experimental: false }
-        - { version: 3.8, experimental: false }
-        - { version: 3.9, experimental: false }
+        python-version: ['3.6', '3.7', '3.8', '3.9']
     steps:
       - uses: actions/checkout@v1
-      - name: Run unit tests
-        shell: bash
-        run: |
-          python --version
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+      - run: python --version
+      - run: python setup.py install
+      - run: pip install -r tests/requirements.txt
+      - run: python -m unittest
```

Comment:
I added unittest command, but some tests are failing. Let me know if the error looks familiar.

Reply:
just click under Details below on the 'Checks', but its not that bad only 2 failures out of 491 tests


```
Ran 491 tests in 26.852s

FAILED (failures=2, errors=3, skipped=8)
```

Reply:
looks like we need to use os.path.join instead of hardcoding the path, since in windows you want \ and in linux you want /, this is on test_bad_config_file1 and test_bad_config_file2. Line 167:

`self.assertIn("The coba configuration file at coba\\tests\\.temp\\.coba", mock_print.call_args_list[1][0][0])`

Reply:
I fixed those, but there's 3 more errors something about env_statistics not being defined.

Reply:
This is awesome! Yeah the environment statistic failures are strange. I'll look into them tomorrow. If I had to guess, that code has some unique 3rd party library dependencies. Coba is built so that 3rd party libraries can be lazy loaded after the fact. To handle this tests with advanced dependencies usually have fall back code in the case that the 3rd party libraries aren't installed. I'd guess some of this fall back logic broke at some point and I didn't catch it because I have all dependencies installed.

There's also a potential issue with a collection of unit tests under test_performance.py. I've always known these tests might cause problems which is why they are all in the same place. All these test calculate the runtime for performance critical code. Of course, that means these tests are prone to fail when running on different machines. These tests failed earlier and look like they are passing now... I'm not sure what the ultimate solution is with these.

Reply:
For now I'm bumping them up, there is an interesting deprecation warning on the python 3.9 run, regarding python 3.10

Reply:
yeap sklearn was missing, I added it and that worked.